### PR TITLE
glass: add poll service

### PR DESCRIPTION
**Pull request by s0nea**
Fixes: https://github.com/aquarist-labs/aquarium/issues/185
Signed-off-by: Tatjana Dehler <tdehler@suse.com>

### CODE REVIEW

**Review by Devp00l — DISMISSED** [review]
LGTM so far :D

File: src/glass/src/app/pages/deployment-page/deployment-page.component.ts [hunk 1]
```diff
@@ -37,43 +36,47 @@ export class DeploymentPageComponent implements OnInit {
...
+    this.startBlockUI('Please wait, fetching device information ...');
+    this.orchService
+      .devices()
       .pipe(
-        delay(5000),
-        mergeMap(() => this.orchService.devices())
+        this.pollService.poll(
+          (hostDevices) => {
+            let hasDevices = false;
+            Object.values(hostDevices).forEach((v) => {
```
Comment (votdev) [inline src/glass/src/app/pages/deployment-page/deployment-page.component.ts:60]: I think this should also work.
```suggestion
            let hasDevices = Object.values(hostDevices).some((v) => v.devices.length);
```

File: src/glass/src/app/pages/deployment-page/deployment-page.component.ts [hunk 2]
```diff
@@ -103,32 +106,26 @@ export class DeploymentPageComponent implements OnInit {
...
 
   pollAssimilationStatus(): void {
-    of(true)
-      .pipe(
-        delay(5000),
-        mergeMap(() => this.orchService.assimilateStatus())
-      )
+    this.orchService
+      .assimilateStatus()
+      .pipe(this.pollService.poll((res) => !res, undefined, 'Failed to deploy disks.'))
```
Comment (votdev) [inline src/glass/src/app/pages/deployment-page/deployment-page.component.ts:121]: We are using `devices` allover the page, so we should use it here, too.
```suggestion
      .pipe(this.pollService.poll((res) => !res, undefined, 'Failed to deploy devices.'))
```

**Review by votdev — CHANGES_REQUESTED** [review]
Works, but need some adaptions.

File: src/glass/src/app/pages/deployment-page/deployment-page.component.ts [hunk 3]
```diff
@@ -37,43 +36,39 @@ export class DeploymentPageComponent implements OnInit {
...
-    of(true)
+  getDevices(): void {
+    this.startBlockUI('Please wait, fetching device information ...');
+    this.orchService
+      .devices()
       .pipe(
-        delay(5000),
-        mergeMap(() => this.orchService.devices())
+        this.pollService.poll(
+          (hostDevices) => Object.values(hostDevices).some((v) => v.devices.length),
```
Comment (votdev) [inline src/glass/src/app/pages/deployment-page/deployment-page.component.ts:58]:
```suggestion
          (hostDevices): boolean => !Object.values(hostDevices).some((v) => v.devices.length),
```

**Review by votdev — CHANGES_REQUESTED** [review]
I would also suggest to move the service to `/shared/services` because it is not related to API only.

**Comment by s0nea** [comment]
> I would also suggest to move the service to `/shared/services` because it is not related to API only.

Yeah... I put it into the `api` folder because the service can't be used outside of it... But if there was/will be a poll API endpoint, it probably makes sense to move it outside.

**Review by votdev — APPROVED** [review]

**Pull request merged**